### ISSUE TITLE
[dumpStack] Dropping >10 items of same kind causes crash

### ISSUE THREAD

**Issue by Abendlied** [Try/OpenGothic#160]
Dropping items from the inventory causes the game to crash and close to desktop.
It appears as thouth the error can only be reproduced if the quantity of the item in the inventory is 10 or more and these are dropped by holding drop key (default is [ALT]).

crash.log attached, see dumpStack(char const*).

[crash.log](https://github.com/Try/OpenGothic/files/6763431/crash.log)

**Comment by Try** [comment]
Hi, @Abendlied and thanks for bug-report!
From surface crash appears to be related to bullet-collision detection:
```
0x000000000060a986: dbg::call_stack<64u>::collect(unsigned int) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x00000000004c449b: CrashLog::dumpStack(char const*) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x00000000004c4ab9: signalHandler(int) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x00000000005e64d7: _gnu_exception_handler in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x00007ffe0d49c776: _C_specific_handler in C:\WINDOWS\SYSTEM32\ntdll.dll
0x00007ffe0d4b217f: _chkstk in C:\WINDOWS\SYSTEM32\ntdll.dll
0x00007ffe0d461454: RtlRaiseException in C:\WINDOWS\SYSTEM32\ntdll.dll
0x00007ffe0d4b0cae: KiUserExceptionDispatcher in C:\WINDOWS\SYSTEM32\ntdll.dll
---- btCollisionDispatcher here ----
0x0000000000569ce7: btCollisionDispatcher::clearManifold(btPersistentManifold*) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x00000000005acbba: btConvexTriangleCallback::btConvexTriangleCallback(btDispatcher*, btCollisionObjectWrapper const*, btCollisionObjectWrapper const*, bool) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x00000000005acc26: btConvexConcaveCollisionAlgorithm::btConvexConcaveCollisionAlgorithm(btCollisionAlgorithmConstructionInfo const&, btCollisionObjectWrapper const*, btCollisionObjectWrapper const*, bool) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x00000000006091cb: btConvexConcaveCollisionAlgorithm::SwappedCreateFunc::CreateCollisionAlgorithm(btCollisionAlgorithmConstructionInfo&, btCollisionObjectWrapper const*, btCollisionObjectWrapper const*) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x0000000000569ad1: btCollisionDispatcher::findAlgorithm(btCollisionObjectWrapper const*, btCollisionObjectWrapper const*, btPersistentManifold*, ebtDispatcherQueryType) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x00000000005697df: btCollisionDispatcher::defaultNearCallback(btBroadphasePair&, btCollisionDispatcher&, btDispatcherInfo const&) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x0000000000606418: btCollisionPairCallback::processOverlap(btBroadphasePair&) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x000000000056397d: btHashedOverlappingPairCache::processAllOverlappingPairs(btOverlapCallback*, btDispatcher*) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x000000000056483b: btHashedOverlappingPairCache::processAllOverlappingPairs(btOverlapCallback*, btDispatcher*, btDispatcherInfo const&) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x000000000056968f: btCollisionDispatcher::dispatchAllCollisionPairs(btOverlappingPairCache*, btDispatcherInfo const&, btDispatcher*) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x000000000056d18a: btCollisionWorld::performDiscreteCollisionDetection() in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x0000000000544928: btDiscreteDynamicsWorld::internalSingleStepSimulation(float) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
0x000000000053ff02: btDiscreteDynamicsWorld::stepSimulation(float, int, float) in D:\Program Files (x86)\Steam\SteamApps\common\Gothic II\OpenGothic\Gothic2Notr.exe
```
I've bee trying to reproduce issue by throwing away items left-and-right - so far no luck.
So any extra info about issue is very welcome.

**Comment by Abendlied** [comment]
It seems I did not test this thouroughly enough, sorry.

I can reliably reproduce this crash in a certain area, when dropping gold coins or mushrooms. I have added a screenshot of the area: beginner area, where Brago hides, near the cave BANCAVE01. What I noted is that the floor there is crooked.
![grafik](https://user-images.githubusercontent.com/86957596/124668283-e86d0000-deb0-11eb-9d2d-008a89141d0e.png)
It also happens inside the cave.
[crash.log](https://github.com/Try/OpenGothic/files/6772891/crash.log)

When I later tried it in town, there was no crash.
On the grass outside of Khorinis it crashed after dumping multiple different items, which all speaks in favour of your collision theory.

**Comment by Try** [comment]
Manage to reproduce it on release build. Meanwhile on recent dev works quite alright - I'll continue to investigate this

**Comment by Abendlied** [comment]
While testing I noticed another, possibly related issue:
If I dump a certain amount of items on the floor and pick them up directly afterwards, the amount of items is reduced.

Example for reproduction:
When I drop mushrooms on the floor (while standing on market place REICH05) I drop 18 mushrooms and pick them up right afterwards. When I check how many I have picked up, my expectation was that it should be 18. But instead I now only have 3 mushrooms.
